dev_roadmap.md

```diff
@@ -0,0 +1,329 @@
+ *main dev*: TBA
+ 
+ Create a new family package qutip-symbolic from ajgpitch fork of Sympy.
+ Must gain permission from Robert Johansson and Eunjong Kim.
+ Extended Sympy simplify to respect non-commuting operators.
+ Produce user documentation.
+ 
+ ### Status messaging and recording
+ &#x1F535; tag: status-msg
+ 
+ *status*: conceptualised
+ 
+ *admin lead*: [Alex](https://github.com/ajgpitch)
+ 
+ *main dev*: TBA
+ 
+ qutip has various ways of recording and reporting status and progress.
+ - `ProgressBar` used by some solvers
+ - Python logging used in qutip.control
+ - `Dump` used in qutip.control
+ - heom records `solver.Stats`
+ 
+ Some consolidation of these would be good.
+ 
+ Some processes (some solvers, correlation, control optimisation) have many stages and many layers. 
+ `Dump` was initially developed to help with debugging, but it is also useful for recording data for analysis.
+ qutip.logging_utils has been criticised for the way it uses Python logging. 
+ The output goes to stderr and hence the output looks like errors in Jupyter notebooks.
+ 
+ Clearly, storing process stage data is costly in terms of memory and cpu time, so any implementation must be able to be optionally switched on/off,
+ and avoided completely in low-level processes (cythonized components). 
+ 
+ Required features:
+ - optional recording (storing) of process stage data (states, operators etc)
+ - optionally write subsets to stdout
+ - maybe other graphical representations
+ - option to save subsets to file
+ - should ideally replace use of `ProgressBar`, Python logging, `control.Dump`, `solver.Stats`
+ 
+ 
+ ### qutip Interactive
+ &#x1F535; tag: qutip-gui
+ 
+ *status*: conceptualised
+ 
+ *admin lead*: [Alex](https://github.com/ajgpitch)
+ 
+ *main dev*: TBA
+ 
+ qutip is pretty simple to use at an entry level for anyone with basic Python skills.
+ However, *some* Python skills are necessary. 
+ A graphical user interface (GUI) for some parts of qutip could help make qutip more accessible.
+ This could be particularly helpful in education, for teachers and learners.
+ 
+ This would make an good GSoC project. It is independent and the scope is flexible.
+ 
+ The scope for this is broad and flexible. Ideas including, but not limited to:
+ 
+ #### Interactive Bloch sphere
+ Matplotlib has some interactive features (sliders, radio buttons, cmd buttons) that can be used to control parameters. 
+ They are a bit clunky to use, but they are there. Could maybe avoid these and develop our own GUI.
+ An interactive Bloch sphere could have sliders for qubit state angles. 
+ Buttons to add states, toggle state evolution path.
+ 
+ #### Interactive solvers
+ Options to configure dynamics generators (Lindbladian / Hamiltonian args etc) and expectation operators.
+ Then run solver and view state evolution.
+ 
+ #### Animated circuits
+ QIP circuits could be animated. Status lights showing evolution of states during the processing.
+ Animated Bloch spheres for qubits. 
```

Comment:
This is definitely a useful feature. But on the other hand, it might need quite some work. Seems that each quantum computing platform has there own developers on this. Because we only have a few developers, I'm wondering if we should compete on this?

Reply:
Good point. I don't think it's high priority